lms/djangoapps/commerce/signals.py

```diff
@@ -69,78 +69,70 @@ def get_request_user():
 
-    try:
-        refund_ids = ecommerce_api_client(request_user or unenrolled_user).refunds.post(
-            {'course_id': course_key_str, 'username': unenrolled_user.username}
-        )
-    except HttpClientError, exc:
-        if exc.response.status_code == 403 and request_user != unenrolled_user:
-            # this is a known limitation; commerce service does not presently
-            # support the case of a non-superusers initiating a refund on
-            # behalf of another user.
-            log.warning("User [%s] was not authorized to initiate a refund for user [%s] "
-                        "upon unenrollment from course [%s]", request_user.id, unenrolled_user.id, course_key_str)
-            return []
-        else:
-            # no other error is anticipated, so re-raise the Exception
-            raise exc
+    service_user = User.objects.get(username=settings.ECOMMERCE_SERVICE_WORKER_USERNAME)
+    api_client = ecommerce_api_client(service_user)
+
+    log.info('Attempting to create a refund for user [%s], course [%s]...', enrollee.id, course_key_str)
+
+    refund_ids = api_client.refunds.post({'course_id': course_key_str, 'username': enrollee.username})
 
     if refund_ids:
-        # at least one refundable order was found.
-        log.info(
-            "Refund successfully opened for user [%s], course [%s]: %r",
-            unenrolled_user.id,
-            course_key_str,
-            refund_ids,
-        )
-
-        # XCOM-371: this is a temporary measure to suppress refund-related email
-        # notifications to students and support@) for free enrollments.  This
-        # condition should be removed when the CourseEnrollment.refundable() logic
-        # is updated to be more correct, or when we implement better handling (and
-        # notifications) in Otto for handling reversal of $0 transactions.
-        if course_enrollment.mode != 'verified':
-            # 'verified' is the only enrollment mode that should presently
-            # result in opening a refund request.
-            log.info(
-                "Skipping refund email notification for non-verified mode for user [%s], course [%s], mode: [%s]",
-                course_enrollment.user.id,
-                course_enrollment.course_id,
-                course_enrollment.mode,
-            )
-        else:
+        log.info('Refund successfully opened for user [%s], course [%s]: %r', enrollee.id, course_key_str, refund_ids)
+
+        failed_refunds = []
+        for refund_id in refund_ids:
             try:
-                send_refund_notification(course_enrollment, refund_ids)
+                # NOTE: Approve payment only because the user has already been unenrolled. Additionally, this
+                # ensures we don't tie up an additional web worker when the E-Commerce Service tries to unenroll
+                # the learner
+                api_client.refunds(refund_id).process.put({'action': 'approve_payment_only'})
```

Comment:
Should references to `payment` here be references to `credit` (e.g., `approve_credit_only)? I can see how it might be conflated with the other meaning of credit on the LMS, but it would be consistent with the vocabulary used on the LMS.